packages/gatsby/src/utils/start-server.ts

```diff
@@ -373,8 +373,8 @@ module.exports = {
 
     // We need the internal webpack file that is used in the bundle, not the module source.
     // It doesn't have the correct sourceMap.
-    const webpackSource = compilation.codeGenerationResults
-      .get(fileModule)
+    const webpackSource = compilation?.codeGenerationResults
```

Comment:
@pieh had an error on this, so some safety guards in case e.g. `fileModule` is undefined

Reply:
My error was
```
Cannot read property 'identifier' of undefined
  TypeError: Cannot read property 'identifier' of undefined
  - CodeGenerationResults.js:33 CodeGenerationResults.get
    [development-runtime]/[webpack]/lib/CodeGenerationResults.js:33:44
  - start-server.ts:377 
    [development-runtime]/[gatsby]/src/utils/start-server.ts:377:8
  - layer.js:95 Layer.handle [as handle_request]
    [development-runtime]/[express]/lib/router/layer.js:95:5
```

So this while good to have, is not handling above error - for that `compilation.codeGenerationResults.get` was defined. It was `fileModule` that was `undefined`

Reply:
You probably want to add this before this line
```
    if (!fileModule) {
      res.json(emptyResponse)
      return
    }
```